mcda/utils/utils_for_main.py

```diff
@@ -322,10 +361,15 @@ def read_matrix(input_matrix_path: str) -> pd.DataFrame:
     :param input_matrix_path: str
     :rtype: pd.DataFrame
     """
+    try:
+        ensure_directory_exists(input_directory_path)
```

Comment:
This method will create the `dirs` if they don't exists. 

Also, from my previous comments:

I think here your code will also try to create folders if it doesn't exists. In my opinion it should not happen in case of input_dir_path. If the dir does not exists then inform the user that input dir is missing and exit the code. Otherwise, you might have to fail the execution at later point of time as there would not be any input files.